fecfile_validate_js/src/index.ts

```diff
@@ -46,7 +46,10 @@ export function validate(
   if (!isValid && !!validator.errors?.length) {
```

Comment:
According to [AJV]( https://ajv.js.org/guide/formats.html), if in this file we import [AJV-Formats](https://github.com/ajv-validator/ajv-formats), we get support for the `date` format.   I think we'd still need to add minLength > 0, for the "" case/

Reply:
@toddlees The issue is not being able to build the fecfile_validate_js distribution when the package is installed from the github repo. Maybe your local setup won't have the problem. In your local fecfile-web-app, change the package.json fecfile-validate depedency entry to point to this commit: 9f2da0395ec3532b7d7d22568e799bd0bc1e461b

Then rm -rf node_modules and package-lock.json and run npm install. When I do this, I get this error and npm install aborts. Maybe this issue is specific to me?:

```
(base) matt@Thermaltake:~/Documents/fec/fecfile-web-app/front-end$ npm install
npm WARN skipping integrity check for git dependency ssh://git@github.com/fecgov/fecfile-validate.git 
npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
npm ERR! code 2
npm ERR! path /home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate
npm ERR! command failed
npm ERR! command sh -c npm run build
npm ERR! > fecfile-validate@0.0.1 build
npm ERR! > rm -rf fecfile_validate_js/dist && tsc --project fecfile_validate_js/tsconfig.json && node fecfile_validate_js/scripts/buildSchemaModules
npm ERR! 
npm ERR! fecfile_validate_js/src/index.ts(28,12): error TS2345: Argument of type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/ajv").default' is not assignable to parameter of type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/core").default'.
npm ERR!   Types of property 'opts' are incompatible.
npm ERR!     Type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/core").InstanceOptions' is not assignable to type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/core").InstanceOptions'.
npm ERR!       Type 'InstanceOptions' is not assignable to type 'CurrentOptions'.
npm ERR!         Types of property 'keywords' are incompatible.
npm ERR!           Type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/types/index").Vocabulary | undefined' is not assignable to type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/types/index").Vocabulary | undefined'.
npm ERR!             Type 'Vocabulary' is not assignable to type '(string | KeywordDefinition)[]'.
npm ERR!               Type 'string | import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/types/index").KeywordDefinition' is not assignable to type 'string | import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/types/index").KeywordDefinition'.
npm ERR!                 Type 'CodeKeywordDefinition' is not assignable to type 'string | KeywordDefinition'.
npm ERR!                   Type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/types/index").CodeKeywordDefinition' is not assignable to type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/types/index").CodeKeywordDefinition'.
npm ERR!                     Types of property 'code' are incompatible.
npm ERR!                       Type '(cxt: import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void' is not assignable to type '(cxt: import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/compile/validate/index").KeywordCxt, ruleType?: string | undefined) => void'.
npm ERR!                         Types of parameters 'cxt' and 'cxt' are incompatible.
npm ERR!                           Type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/compile/validate/index").KeywordCxt' is not assignable to type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/compile/validate/index").KeywordCxt'.
npm ERR!                             The types of 'gen._scope' are incompatible between these types.
npm ERR!                               Type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/ajv/dist/compile/codegen/scope").Scope' is not assignable to type 'import("/home/matt/Documents/fec/fecfile-web-app/front-end/node_modules/fecfile-validate/node_modules/ajv/dist/compile/codegen/scope").Scope'.
npm ERR!                                 Property '_names' is protected but type 'Scope' is not a class derived from 'Scope'.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/matt/.npm/_logs/2022-06-24T11_03_49_032Z-debug-0.log

```